ingenialink/servo.py

```diff
@@ -273,6 +275,65 @@ def is_listener_started(self) -> bool:
         """
         return self.__listener_servo_status is not None
 
+    def check_configuration(self, config_file: str, subnode: Optional[int] = None) -> None:
+        """Check if the drive is configured in the same way as the given configuration file.
+        Compares the value of each register in the given file with the corresponding value in the
+        drive.
+
+        Args:
+            config_file: the configuration to check
+            subnode: Subnode of the axis. Defaults to None.
+
+        Raises:
+            FileNotFoundError: If the configuration file cannot be found.
+            ValueError: If a configuration file from a subnode different from 0
+                is attempted to be loaded to subnode 0.
+            ValueError: If an invalid subnode is provided.
+            ILConfigurationError: If the configuration file differs from the drive state.
+
+        """
+        if subnode is not None and (not isinstance(subnode, int) or subnode < 0):
+            raise ValueError("Invalid subnode")
+        _, registers = self._read_configuration_file(config_file)
+
+        dest_subnodes = set(int(element.attrib["subnode"]) for element in registers)
+        if subnode == 0 and subnode not in dest_subnodes:
+            raise ValueError(f"Cannot load {config_file} to subnode {subnode}")
+        cast_data = {"float": float, "str": str}
+        registers_errored = []
+        for element in registers:
+            if "storage" in element.attrib:
+                if subnode is None:
+                    element_subnode = int(element.attrib["subnode"])
+                else:
+                    element_subnode = subnode
+                reg_dtype = element.attrib["dtype"]
+                reg_data = element.attrib["storage"]
+                reg_id = element.attrib["id"]
+                try:
+                    stored_data = self.read(reg_id, element_subnode)
+                    reg_data = cast_data.get(reg_dtype, int)(reg_data)
+                    # Precision is set so at least the first four decimals have to be equal
+                    if not np.isclose(reg_data, stored_data, rtol=1e-6):
+                        registers_errored.append(
+                            f"{reg_id} --- Expected: {reg_data} | Found: {stored_data}\n"
+                        )
+                except ILError as e:
+                    reg_id = element.attrib["id"]
+                    il_error = (f"{reg_id} -- {e}",)
+                    logger.error(
+                        il_error,
+                        str(element.attrib["id"]),
+                        e,
+                    )
+                    registers_errored.append(il_error)
```

Comment:
I'm confused about the typing of registers_errored.
Is i a list of strings and tuples at the same time?